Removed feedforward_jitter_factor and feedforward_averaging froom TUNE

limonspb opened the pull request:
Both `feedforward_jitter_factor` and `feedforward_averaging` are parts of the RC_LINK category
so TUNEs don't need and can't set these values. Unless inside the RC_LINK options.
The final CLI outcome must be the same regardless of the order that user applies presets (RC_LINK + TUNE == TUNE + RC_LINK)

So i removed `feedforward_jitter_factor` from tune/defaults.txt
removed `feedforward_jitter_factor` and `feedforward_averaging` from the tunes
replaced old RC_LINK options in the tunes with the proper `#$ INCLUDE rc_link/....`

Please check the changes with care.

limonspb commented:
[rochford77](https://github.com/rochford77)
[FPVian](https://github.com/FPVian)

SupaflyFPV commented:
Jitter factor has a dual purpose - it was designed to deal with jitter but it has now taken over from Feedforward transition. It is used as a tuning solution to soften the action of feedforward so I think it should be maintained also for a TUNE parameter as well as RC LINK...

limonspb commented:
> Jitter factor has a dual purpose - it was designed to deal with jitter but it has also now taken over from Feedforward transition. It is used as a tuning solution to soften the action of feedforward and is an important tuning parameter now - so I think it should be maintained also for a TUNE parameter as well as RC LINK...is there a reason it cannot exist in both categories?

Existing in both categories means the final value for the user is unknown. Today user applies TUNE first, then RC_link, tomorrow for the second quad he applies RC_LINK first and the TUNE.

SupaflyFPV commented:
> > Jitter factor has a dual purpose - it was designed to deal with jitter but it has also now taken over from Feedforward transition. It is used as a tuning solution to soften the action of feedforward and is an important tuning parameter now - so I think it should be maintained also for a TUNE parameter as well as RC LINK...is there a reason it cannot exist in both categories?
> 
> Existing in both categories means the final value for the user is unknown. Today user applies TUNE first, then RC_link, tomorrow for the second quad he applies RC_LINK first and the TUNE.

Ok I see the issue...

Can you give me some time to explore the RC Link categorisation/organisation because currently I would prefer to keep:

Jitter factor
FF Boost

...in the TUNE categories seeing as it influences the feel and behaviour which is part of the 'seasoning' for the tunes...

I see some 'feel based' categories such as 'Freestyle, Cinematic' etc are existing in the RC LINK settings...

For instance there is an argument for aggressive freestyle to have low RC Smoothing (and lower RC latency like racer) but high jitter factor (less aggressive FF) - not currently an option....

@ctzsnooze is away I think unfortunately....

limonspb commented:
> > > Jitter factor has a dual purpose - it was designed to deal with jitter but it has also now taken over from Feedforward transition. It is used as a tuning solution to soften the action of feedforward and is an important tuning parameter now - so I think it should be maintained also for a TUNE parameter as well as RC LINK...is there a reason it cannot exist in both categories?
> > 
> > 
> > Existing in both categories means the final value for the user is unknown. Today user applies TUNE first, then RC_link, tomorrow for the second quad he applies RC_LINK first and the TUNE.
> 
> Ok I see the issue...
> 
> Can you give me some time to explore the RC Link categorisation/organisation because currently I would prefer to keep:
> 
> Jitter factor FF Boost
> 
> ...in the TUNE categories (as well) seeing as it influences the feel and behaviour which is part of the 'seasoning' for the tunes...
> 
> I see some 'feel based' categories such as 'Freestyle, Cinematic' etc are existing in the RC LINK settings...
> 
> ...but there is nothing for the scenario of aggressive freestyle - to have low RC Smoothing (and lower RC latency like racer) but high jitter factor (less aggressive FF)....
> 
> @ctzsnooze is away I think unfortunately....

ff_boost should be removed from TUNE too to guarantee the outcome, well, in the next PR.
we have 5 RC_LINK groups:
race, freestyle, HD_freestyle, cinematic, ultra_cinematic. This should be enough i think.
But we can make a new rc_link "aggressive freestyle" if we really need it. Then also "smooth_racing" and then "soft_freestyle" and then "aggressive_cinematic" :))

SupaflyFPV commented:
to add to the previous discussion, these mentioned duplicated parameter settings in dispute are also in TUNE presets:

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/ctzsnooze_5in_4S_race.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/karate_freestyle_5_inch.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/karate_race.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/tiny_karate.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/fpvian_basher_freestyle.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/mouseFPV_AOS_35_tune_filters.txt

https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/whoop_justice.txt

....

Karate Race also does separate RC settings as option - which could be a good potential solution....

limonspb commented:
> to add to the previous discussion, these mentioned duplicated parameter settings in dispute are also in TUNE presets:
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/ctzsnooze_5in_4S_race.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/karate_freestyle_5_inch.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/karate_race.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/tiny_karate.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/fpvian_basher_freestyle.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/mouseFPV_AOS_35_tune_filters.txt
> 
> https://github.com/betaflight/firmware-presets/blob/master/presets/4.3/tune/whoop_justice.txt
> 
> ....
> 
> Karate Race also does separate RC settings as option - which could be a good potential solution....

Yes in this PR I fixed them all, and replaced some old RC link options with the new ones (in the past rc_link didn't include rc_smoothig)

SupaflyFPV commented:
Ok......as far as the PR is concerned I'm fine for the changes laid out (removing jitter factor in my presets). Maybe the others have some comments. As mentioned as far as the preset functionality is concerned I see the issue requires fixing.

In my next tune update I will consider the inclusion of RC Link parameters as options, or with CTZ we can discuss the RC Link categories.

sugaarK reviewed (approved):
I have to agree with @limonspb its very much a  flying style/radio link/rc smoothing function. Until we have the settings auto configurator to suit the rc link speed it’s very link dependant

SupaflyFPV commented:
ok. Maybe the positioning of this parameter has been changed since I last was involved. The crossover nature Chris explained to me is outlined in the tuning notes:

https://github.com/betaflight/betaflight/wiki/4.3-Tuning-Notes#Feedforward-jitter-reduction

FPVian commented:
Thanks for tagging me! In the future it would be nice to bring these back to tuning presets, but the way things are today, I agree it makes more sense to keep them exclusive to RC_LINK.
